### PR TITLE
Update automation scripts and docs

### PR DESCRIPTION
**Pull request by backwind1233**
## Context
Linked to https://github.com/oracle/weblogic-kubernetes-operator/issues/4352

## Goal
- This PR is used to update the automation scripts [create-domain-on-aks.sh](https://github.com/oracle/weblogic-kubernetes-operator/blob/0e374fee6742a4a915306205e06a57d30e422fbe/kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh), to make it work as expected.
- Update the doc: domain-on-pv.md 

I also create a github [[workflow](https://github.com/azure-javaee/weblogic-kubernetes-operator/actions/workflows/oracal-wko-sample-automation.yaml)] used to verify the automation scripts `create-domain-on-aks.sh`.

## Feature or Improvments
- The automation script are using `v4.1.1`
- For now, users can just run `create-domain-on-aks.sh` directly.
- The automation script will prompt the user to enter the necessary parameters.
- The automation script will validate the host machine meets environment requirements, like java installed or not, az cli login or not.
  - image

- The automation script will check some parameter at the beginning, like the oracle account.
  - image

## Test
✅ Test passed; Check within this [workflow](https://github.com/azure-javaee/weblogic-kubernetes-operator/actions/runs/6105883744/attempts/1).
image

**Workflow status**
[WKO AKS Sample Auto Workflow](https://github.com/azure-javaee/weblogic-kubernetes-operator/actions/workflows/oracal-wko-sample-automation.yaml)

### CODE REVIEW

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+export dockerEmail=docker-email
+
+# Oracle SSO account password, used to pull the WebLogic Server image.
+export dockerPassword=docker-password
+
+# Name of weblogic user account.
+export weblogicUserName=weblogic
+
+# Password for WebLogic user account.
+export weblogicAccountPassword=Secret123456
```
Comment (mriccell) [inline kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh:21]: Are these hard coded secrets?

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+#
+# Parameters that must be changed from these values!
+#
+
+# Oracle Single Sign-On (SSO) account email, used to pull the WebLogic Server image.
+export dockerEmail=docker-email
+
+# Oracle SSO account password, used to pull the WebLogic Server image.
+export dockerPassword=docker-password
```
Comment (mriccell) [inline kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh:15]: is this a hard coded password?

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+# Kubernetes cluster is named with ${namePrefix}akscluster<timestamp>, e.g. wlsakscluster1592469388
+# Storage account is named with ${namePrefix}storage<timestamp>, e.g. wlsstorage1592469388
+export namePrefix=wls
+
+#
+# Parameters that may optionally be changed.
+#
+
+# The suffix of file share secret name, the complete value is ${namePrefix}${azureFileShareSecretNameSuffix}.
+export azureFileShareSecretNameSuffix=azure-secret
```
Comment (mriccell) [inline kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh:37]: Is this a hard coded secret?

**Review by mriccell — CHANGES_REQUESTED** [review]

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh (hunk 2 above)
Comment (backwind1233) [reply kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh:15]: Even though the file `create-domain-on-aks-inputs.sh` has a `.sh` extension, its main purpose is to serve as a **configuration** file. 
Users can modify certain configuration options within the file, and if they don't modify the parameters, the values within it serve as default values during startup.

**Comment by edburns** [comment]
Hello @backwind1233 , I am pretty sure the following issue is present in the content before you came to it. Nonetheless, I need to see it addressed.

In the section ** Prepare parameters** there is a code block that includes

```bash
# An example of Domain_Creation_Image_URL: xxx.azurecr.io/wdt-domain-image:WLS-v1
export Domain_Creation_Image_URL=<replace with your domain creation image url>
```

The reader has not even yet created the cluster yet. I will re-work the order as I continue my review tomorrow.

File: documentation/site/content/samples/azure-kubernetes-service/includes/create-aks-cluster-storage.txt [hunk 4]
```diff
@@ -184,91 +181,65 @@ You will dynamically create and use a persistent volume with Azure Files NFS sha
...
-
-**NOTE**: Beyond the required and default configurations generated by the command, you can modify the generated YAML files to further customize your deployment. For further information about customizing your deployment, consult the operator documentation, [AKS documentation](https://docs.microsoft.com/azure/aks/), and Kubernetes references.
-
-##### Apply generated configuration files
-
-To define how an Azure file NFS share is created, we have provided a configuration file `azure-csi-nfs.yaml`. You can find it in your output directory.
-
-The following command displays the YAML content of your current settings. This should be the same with content in `azure-csi-nfs.yaml`. The Storage Class name is `azurefile-csi-nfs`.
+cat >azure-csi-nfs-${TIMESTAMP}.yaml <<EOF
+# Copyright (c) 2018, 2021, Oracle and/or its affiliates.
```
Comment (jshum2479) [inline documentation/site/content/samples/azure-kubernetes-service/includes/create-aks-cluster-storage.txt:191]: Copyright year 2023.  Do we even need it in this type of situation?

File: documentation/site/content/samples/azure-kubernetes-service/includes/create-aks-cluster-storage.txt [hunk 5]
```diff
@@ -184,91 +181,65 @@ You will dynamically create and use a persistent volume with Azure Files NFS sha
...
-persistentvolume/wls-azurefile created
+$ kubectl apply -f azure-csi-nfs-${TIMESTAMP}.yaml
+$ kubectl apply -f pvc-${TIMESTAMP}.yaml
 ```
 
 Use the following command to verify:
 
 ```shell
-$ kubectl get sc
+$ kubectl get pvc
```
Comment (jshum2479) [inline documentation/site/content/samples/azure-kubernetes-service/includes/create-aks-cluster-storage.txt:236]: The above commands in 229-230 create both sc and pvc,   shouldn't we check both withe kubectl get ? (sc is removed)

Comment (backwind1233) [reply documentation/site/content/samples/azure-kubernetes-service/includes/create-aks-cluster-storage.txt:236]: Yes, it's better to check both, will update.

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-generate-yaml.sh [hunk 6]
```diff
@@ -35,7 +35,7 @@ spec:
 
   # Identify which Secret contains the credentials for pulling an image
   imagePullSecrets:
-    - name: ${namePrefix}regcred
+    - name: ${NAME_PREFIX}regcred
```
Comment (edburns) [inline kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-generate-yaml.sh:38]: Thanks. I wondered about that.

File: documentation/site/content/samples/azure-kubernetes-service/domain-on-pv.md [hunk 7]
```diff
@@ -307,7 +309,7 @@ The domain resource references the cluster resource, a WebLogic Server installat
   If the WLS Administration Console is still not available, use `kubectl get events --sort-by='.metadata.creationTimestamp' ` to troubleshoot.
 
   ```shell
-  $ kubectl get events --sort-by='.metadata.creationTimestamp
+  $ kubectl get events --sort-by='.metadata.creationTimestamp'
```
Comment (edburns) [inline documentation/site/content/samples/azure-kubernetes-service/domain-on-pv.md:312]: Good catch.

**Review by edburns — APPROVED** [review]

File: documentation/site/content/samples/azure-kubernetes-service/domain-on-pv.md [hunk 8]
```diff
@@ -74,7 +74,7 @@ export Domain_Creation_Image_tag=<replace with the Docker tag for the WDT image
 
 # Specify a prefix to name resources, only allow lowercase letters and numbers, between 1 and 7 characters
 export BASE_DIR=~
-export namePrefix=wls
+export NAME_PREFIX=wls
```
Comment (edburns) [inline documentation/site/content/samples/azure-kubernetes-service/domain-on-pv.md]: Thanks.

File: documentation/site/content/samples/azure-kubernetes-service/domain-on-pv.md [hunk 9]
```diff
@@ -202,6 +202,14 @@ wlsregcred                                kubernetes.io/dockerconfigjson   1
...
 **NOTE**: If the `NAME` column in your output is missing any of the values shown above, please review your execution of the preceding steps in this sample to ensure that you correctly followed all of them.
 
+##### Enable Weblogic Operator
+
+Run the following command to enable the operator to monitor the namespace.
+
+```shell
+kubectl label namespace default weblogic-operator=enabled
+```
+
```
Comment (edburns) [inline documentation/site/content/samples/azure-kubernetes-service/domain-on-pv.md]: I am glad we caught this important step.

**Review by edburns — APPROVED** [review]

**Review by edburns — APPROVED** [review]

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh (hunk 1 above)
Comment (edburns) [reply kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh:21]: No, it is not. See below.

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh (hunk 2 above)
Comment (edburns) [reply kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh:15]: No, it is not a hard coded password. The user must change it.

File: kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh (hunk 3 above)
Comment (edburns) [reply kubernetes/samples/scripts/create-weblogic-domain-on-azure-kubernetes-service/create-domain-on-aks-inputs.sh:37]: no, this is the name of the secret, not the secret value.

**Review by edburns — APPROVED** [review]

**Review by mriccell — APPROVED** [review]

**Comment by rjeberhard** [comment]
@backwind1233 @edburns @mriccell, just FYI... As part of our new automation to accept external contributions, I've pulled this branch to an Oracle-internal git instance and created a peer MR there. I'll merge that and then a different automation will push commits back from the internal git to this GitHub project. I'm sorry that this means that I'll ultimately just close this PR, but we've been backed in to this because of requirements from Oracle Global Security.

I've had to make a few minor edits:
* I've removed the inadvertent change to add an import to a integration test file. This is obviously not intentional.
* We have validation of scripts that runs as part of the build and so to satisfy that validation, I've:
  * Removed directory use of `kubectl` and replaced with a variable
  * Updated the exclusion list to allow for the uses of the "docker" literal.
  * Removed use of the Bash `function` keyword to support more versions of Bash.

**Pull request merged**